pydicer/convert/convert_pt.py

```diff
@@ -0,0 +1,311 @@
+        raise RuntimeError("A slice comes from another serie")
+
+    axial_positions = np.asarray([k.ImagePositionPatient[2] for k in slices])
+    # Compute redundant slice positions
+    ind2rm = [
+        ind
+        for ind in range(len(axial_positions))
+        if axial_positions[ind] == axial_positions[ind - 1]
+    ]
+    # Check if there is redundancy in slice positions and remove them
+    if len(ind2rm) > 0:
+        slices = [k for i, k in enumerate(slices) if i not in ind2rm]
+        axial_positions = np.asarray([k.ImagePositionPatient[2] for k in slices])
+
+    slice_spacing = (
+        slices[1].ImagePositionPatient[2] - slices[0].ImagePositionPatient[2]
+    )
+
+    pixel_spacing = np.asarray(
+        [
+            slices[0].PixelSpacing[0],
+            slices[0].PixelSpacing[1],
+            slice_spacing,
+        ]
+    )
+
+    if modality == "PT":
+        np_image = get_physical_values_pt(slices, patient_weight, dtype=dtype_image)
+    else:
+        raise ValueError("The modality {} is not supported".format(modality))
+
+    position_final_slice = (len(slices) - 1) * slice_spacing + slices[
+        0
+    ].ImagePositionPatient[2]
+    # Test whether some slices are missing
+    # Pang -- due to an error at line 144: TypeError: only size-1 arrays can be converted
+    # to Python scalars
+    if not is_approx_equal(
+            position_final_slice, float(slices[-1].ImagePositionPatient[2])
+        ):
+        if (position_final_slice - axial_positions[-1]) / slice_spacing < 1.5:
+            # If only one slice is missing
+            diff = np.asarray(
+                [
+                    not is_approx_equal(
+                        float(axial_positions[ind])
+                        - float(axial_positions[ind - 1])
+                        - slice_spacing,
+                        0,
+                    )
+                    for ind in range(1, len(axial_positions))
+                ]
+            )
+            ind2interp = int(np.where(diff)[0])
+            new_slice = (
+                np_image[:, :, ind2interp] + np_image[:, :, ind2interp + 1]
+            ) * 0.5
+            new_slice = new_slice[..., np.newaxis]
+            np_image = np.concatenate(
+                (np_image[..., :ind2interp], new_slice, np_image[..., ind2interp:]),
+                axis=2,
+            )
+            warnings.warn(
+                "One slice is missing, we replaced it by linear interpolation"
+            )
+        else:
+            # if more than one slice are missing
+            raise RuntimeError("Multiple slices are missing")
+
+    image_position_patient = [float(k) for k in slices[0].ImagePositionPatient]
+    sitk_image = get_sitk_volume_from_np(
+        np_image, pixel_spacing, image_position_patient
+    )
+
+    sitk_writer.SetFileName(output_filepath)
+    sitk_writer.Execute(sitk_image)
+
+    return np.transpose(np_image, (1, 0, 2)), pixel_spacing, image_position_patient

```

Comment:
Why are we returning this? I would suggest returning a path to the output nifti... The user of this function can then load it and read whatever they want from it. Alternatively, we could return the sitk.Image here. But returning a np.array seems like this was implemented for some other specific use case here.